mods/persistence/game/machinery/cryopod.dm

```diff
@@ -75,18 +70,57 @@
 
 // Players shoved into this will be removed from the game and added to limbo to be deserialized later.
 /obj/machinery/cryopod/despawner
-	name = "experimental cryopod"
-	time_till_despawn = 20
+	name = "cryo storage pod"
+	time_till_despawn = 60 SECONDS
 
-/obj/machinery/cryopod/despawner/Process()
+//#FIXME: This messing with spawnlists could definitely be improved on.
+/obj/machinery/cryopod/despawner/Initialize()
 	. = ..()
-	if ((world.time - time_entered < time_till_despawn) && (occupant.ckey))
-		return
-	despawn_occupant()
+	global.latejoin_locations |= get_turf(src)
+	global.latejoin_cryo_locations |= get_turf(src)
+
+/obj/machinery/cryopod/despawner/forceMove(atom/dest)
+	global.latejoin_locations -= get_turf(src)
+	global.latejoin_cryo_locations -= get_turf(src)
+	. = ..()
+	global.latejoin_locations |= get_turf(src)
+	global.latejoin_cryo_locations |= get_turf(src)
+
+/obj/machinery/cryopod/despawner/Destroy()
+	global.latejoin_locations -= get_turf(src)
+	global.latejoin_cryo_locations -= get_turf(src)
+	. = ..()
+
+/obj/machinery/cryopod/despawner/Process()
+	if(occupant)
+		if(applies_stasis && iscarbon(occupant) && (world.time > time_entered + 20 SECONDS))
+			var/mob/living/carbon/C = occupant
+			C.SetStasis(2)
+
+		var/time_elapsed = world.time - time_entered
+		var/time_left = round((time_till_despawn - time_elapsed) / (1 SECOND)) 
+		if((time_left > 0) && ((time_left % 5) == 0))
+			to_chat(occupant, SPAN_NOTICE("[time_left] seconds left until transfer to deep storage.."))
+		
+		//Force despawn when no client
+		if ((time_elapsed < time_till_despawn) && occupant.ckey)
+			return
+		spawn(1)
+			despawn_occupant()
 
 /obj/machinery/cryopod/despawner/despawn_occupant()
+	set waitfor = FALSE
```

Comment:
Little bit unsure on this. What was the reason for this change?

Reply:
Oh, that's because I tried adding some sleeps in the despawn procs to send some timed messages. And I tried to get dreamchecker to understand it was not gonna block the entire process stack. Sadly dreamchecker never understood so I forgot about those...